backend_new/src/services/listing.service.ts

```diff
@@ -0,0 +1,330 @@
+            })),
+          });
+        } else if (ListingFilterKeys.bedrooms in filter) {
+          const builtFilter = buildFilter({
+            $comparison: filter.$comparison,
+            $include_nulls: false,
+            value: filter[ListingFilterKeys.bedrooms],
+            key: ListingFilterKeys.bedrooms,
+          });
+          filters.push({
+            OR: builtFilter.map((filt) => ({
+              units: {
+                some: {
+                  numBedrooms: filt,
+                },
+              },
+            })),
+          });
+        } else if (ListingFilterKeys.zipcode in filter) {
+          const builtFilter = buildFilter({
+            $comparison: filter.$comparison,
+            $include_nulls: false,
+            value: filter[ListingFilterKeys.zipcode],
+            key: ListingFilterKeys.zipcode,
+          });
+          filters.push({
+            OR: builtFilter.map((filt) => ({
+              listingsBuildingAddress: {
+                zipCode: filt,
+              },
+            })),
+          });
+        } else if (ListingFilterKeys.leasingAgents in filter) {
+          const builtFilter = buildFilter({
+            $comparison: filter.$comparison,
+            $include_nulls: false,
+            value: filter[ListingFilterKeys.leasingAgents],
+            key: ListingFilterKeys.leasingAgents,
+          });
+          filters.push({
+            OR: builtFilter.map((filt) => ({
+              userAccounts: {
+                some: {
+                  id: filt,
+                },
+              },
+            })),
+          });
+        } else if (ListingFilterKeys.jurisdiction in filter) {
+          const builtFilter = buildFilter({
+            $comparison: filter.$comparison,
+            $include_nulls: false,
+            value: filter[ListingFilterKeys.jurisdiction],
+            key: ListingFilterKeys.jurisdiction,
+          });
+          filters.push({
+            OR: builtFilter.map((filt) => ({
+              jurisdictionId: filt,
+            })),
+          });
+        }
+      });
+    }
+
+    if (search) {
+      filters.push({
+        name: {
+          contains: search,
+          mode: Prisma.QueryMode.insensitive,
+        },
+      });
+    }
+
+    return {
+      AND: filters,
+    };
+  }
+
+  /*
+    this will return 1 listing or error
+    the scope of data it returns is dependent on the view arg passed in
+  */
+  async findOne(
+    listingId: string,
+    lang: LanguagesEnum = LanguagesEnum.en,
+    view = 'full',
+  ) {
+    const listingRaw = await this.prisma.listings.findFirst({
+      include: views[view],
+      where: {
+        id: {
+          equals: listingId,
+        },
+      },
+    });
+
+    const result = mapTo(ListingGet, listingRaw);
+
+    if (!result) {
+      throw new NotFoundException();
+    }
+
+    if (lang !== LanguagesEnum.en) {
+      // TODO: await this.translationService.translateListing(result, lang);
```

Comment:
just making sure you still need this

Reply:
I do yea, that translationService hasn't been prisma-ified yet :(